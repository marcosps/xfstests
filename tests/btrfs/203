#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright (C) 2020 SUSE Linux Products GmbH. All Rights Reserved.
#
# FSQA Test No. 203
#
# Test subvolume deletion using the subvolume id, even when the subvolume in
# question is in a different mount space.
#
seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"
tmp=/tmp/$$
status=1	# failure is the default!

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/filter.btrfs

# real QA test starts here
_supported_fs btrfs
_supported_os Linux
_require_scratch
_require_btrfs_command subvolume delete --subvolid

_scratch_mkfs > /dev/null 2>&1
_scratch_mount

# Test creating a normal subvolumes
_run_btrfs_util_prog subvolume create $SCRATCH_MNT/subvol1 | _filter_scratch
_run_btrfs_util_prog subvolume create $SCRATCH_MNT/subvol2 | _filter_scratch
_run_btrfs_util_prog subvolume create $SCRATCH_MNT/subvol3 | _filter_scratch

echo "Current subvolume ids:"
$BTRFS_UTIL_PROG subvolume list $SCRATCH_MNT | awk '{ print $NF }'

# Delete the subvolume subvol1 using it's subvolume id
SUBVOLID=$(_btrfs_get_subvolid $SCRATCH_MNT subvol1)
$BTRFS_UTIL_PROG subvolume delete --subvolid $SUBVOLID  $SCRATCH_MNT | _filter_scratch

echo "After deleting one subvolume:"
# should present only two subvolumes
$BTRFS_UTIL_PROG subvolume list $SCRATCH_MNT | awk '{ print $NF }'

umount $SCRATCH_MNT

# Now we mount the subvol2, which makes subvol3 not accessible for this mount
# point, but we should be able to delete it using it's subvolume id
$MOUNT_PROG -o subvol=subvol2 $SCRATCH_DEV $SCRATCH_MNT
SUBVOLID=$(_btrfs_get_subvolid $SCRATCH_MNT subvol3)
$BTRFS_UTIL_PROG subvolume delete --subvolid $SUBVOLID $SCRATCH_MNT | _filter_scratch

echo "Last remaining subvolume:"
$BTRFS_UTIL_PROG subvolume list $SCRATCH_MNT | awk '{ print $NF }'

umount $SCRATCH_MNT

# now mount the rootfs
_scratch_mount

# Delete the subvol2
SUBVOLID=$(_btrfs_get_subvolid $SCRATCH_MNT subvol2)
$BTRFS_UTIL_PROG subvolume delete --subvolid $SUBVOLID  $SCRATCH_MNT | _filter_scratch

echo "All subvolumes removed."
$BTRFS_UTIL_PROG subvolume list $SCRATCH_MNT | awk '{ print $NF }'

umount $SCRATCH_MNT

# success, all done
status=0
exit
