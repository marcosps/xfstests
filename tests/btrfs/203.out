QA output created by 203
256
258
259
Delete subvolume (no-commit): 'SCRATCH_MNT/subvol1'
258
259
Delete subvolume (no-commit): 'SCRATCH_MNT/subvol3'
258
Delete subvolume (no-commit): 'SCRATCH_MNT/subvol2'
